Source/ACE/Factories/WorldObjectFactory.cs

```diff
@@ -83,11 +84,30 @@ public static WorldObject CreateWorldObject(AceObject aceO)
                     return new Game(aceO);
                 case WeenieType.GamePiece:
                     return new GamePiece(aceO);
+                case WeenieType.Clothing:
+                    return new Clothing(aceO);
+                case WeenieType.Missile:
+                    return new Missile(aceO);
+                case WeenieType.Ammunition:
+                    return new Ammunition(aceO);
+                case WeenieType.MeleeWeapon:
+                    return new MeleeWeapon(aceO);
+                case WeenieType.Caster:
+                    return new Caster(aceO);
                 default:
                     return new GenericObject(aceO);
             }
         }
 
+        public static WorldObject CreateCorpse(Creature cs)
+        {
+            ////Console.WriteLine("Inside CreateCorpse");
+            WorldObject wo = CreateNewWorldObject(1217);
+            ////Console.WriteLine("Added location to corpse");
+            wo.Location = cs.Location;
+            return wo;
+        }
+
```

Comment:
Corpse creation should probably be moved to its own WO class that is derived from Container, instead of being a method within the WorldObjectFactory class.